src/icp_canister_backend/tests/deposit_tests.rs

```diff
@@ -63,32 +43,20 @@ fn test_deposit_fails_without_transfer() {
 
 #[test]
 fn test_shares_calculation() {
-    let (pic, canister_id, ledger_id) = setup();
+    let s = setup();
+    let mut client = s.client();
     let user1 = Principal::from_text("xkbqi-2qaaa-aaaah-qbpqq-cai").unwrap();
 
-    let current_episode = get_stakable_episode(&pic, canister_id, 0);
+    let current_episode = client.get_stakable_episode(0);
 
     let deposit_amount = Nat::from(200_000_000u64);
 
     // First user deposits
-    create_deposit(
-        &pic,
-        canister_id,
-        ledger_id,
-        user1,
-        deposit_amount.clone(),
-        current_episode,
-    );
-
-    let pool_state_result = pic
-        .query_call(
-            canister_id,
-            user1,
-            "get_pool_state",
-            encode_args(()).unwrap(),
-        )
-        .expect("Failed to get pool state");
-    let pool_state: PoolState = decode_one(&pool_state_result).unwrap();
+    client
+        .connect(user1)
+        .create_deposit(user1, deposit_amount.clone(), current_episode);
```

Comment:
Create deposit shouldn't accept user parameters as it is already connected in the client.
However, it looks like I am wrong about putting this function into the client in general.
We need to have a create_deposit as util function that accepts client struct as a parameter.
So, create_deposit(client, amount, episode)
The same thing for the reward_pool util function, get_stakable_episode, and other utils function.

The idea is that the client struct simplifies all of the canister methods but doesn't include helpers and utils as they are not the responsibility of this struct. Additional logic should be implemented separately.
Also, rename Clien struct to PoolCanister. When we have more canisters, we will need to have a separate struct for each of them.